src/Type.cc

```diff
@@ -1337,6 +1337,10 @@ void RecordType::Create(std::vector<std::optional<ZVal>>& r) const
 		{
 		auto& init = field_inits[i];
 
+		if ( ! init )
```

Comment:
This doesn't quite seem the right fix: IIUC, elements of `field_inits` are never NULL. `fields_inits` in the test case has size 1 and the index used is `1` which is overreading the vector. It just happens to yield NULL. Changing `[i]` to `at(i)` raises `std::out_of_range`:
```
terminate called after throwing an instance of 'std::out_of_range'
  what():  vector::_M_range_check: __n (which is 1) >= this->size() (which is 1)
```

Further, in the same realm, the following test case triggers an assertion:
```
# @TEST-EXEC-FAIL: zeek -b %INPUT
# @TEST-EXEC: TEST_DIFF_CANONIFIER=$SCRIPTS/diff-remove-abspath btest-diff .stderr

type test: record {
        a: count;
        a: count;
        b: count;
};

global a: test;
```
error in ./language/record-duplicate-fields-global.zeek, line 8: duplicate field 'a' found in record definition
zeek: ../src/Type.cc:1062: void zeek::RecordType::AddField(unsigned int, const zeek::TypeDecl*): Assertion `field == field_inits.size()' failed.
```